packages/upload-api/src/types/blob.ts

```diff
@@ -0,0 +1,74 @@
+import type {
+  UnknownLink,
+  Invocation,
+  Result,
+  Failure,
+  DID,
+} from '@ucanto/interface'
+import { BlobMultihash, BlobListItem } from '@web3-storage/capabilities/types'
+
+import { RecordKeyConflict, ListOptions, ListResponse } from '../types.js'
+import { Storage } from './storage.js'
+
+export type TasksStorage = Storage<UnknownLink, Invocation>
+
+export interface AllocationsStorage {
+  get: (
+    space: DID,
+    blobMultihash: BlobMultihash
+  ) => Promise<Result<BlobGetOutput, Failure>>
+  exists: (
+    space: DID,
+    blobMultihash: BlobMultihash
+  ) => Promise<Result<boolean, Failure>>
+  /** Inserts an item in the table if it does not already exist. */
+  insert: (
+    item: BlobAddInput
+  ) => Promise<Result<BlobAddOutput, RecordKeyConflict>>
+  list: (
+    space: DID,
+    options?: ListOptions
+  ) => Promise<Result<ListResponse<BlobListItem>, Failure>>
+}
+
+export interface Blob {
```

Comment:
Don't call it `Blob` please as TS sometimes tries to unify it with builtin `Blob` interface, maybe call it `Content` or `BlobModel` instead.